### DISCUSSION TITLE
playing with trimesh (+ shapely)

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#228] [Show and tell]
![image](https://user-images.githubusercontent.com/3694604/216197226-d37cb3c2-3cc7-4c5a-9018-ec2d0aee911b.png)

```python
import trimesh
import shapely
# https://iamkate.com/data/12-bit-rainbow/
palette = (
    '#817', '#a35', '#c66', '#e94',
    '#ed0', '#9d5', '#4d8', '#2cb',
    '#0bc', '#09c', '#36b', '#639'
    )

def setup():
    global m
    size(400, 400, P3D)
    no_stroke()
    polygon = shapely.geometry.Polygon([(-100, -100), (0, -100),
                                        (0, 0), (-50, -50), (-100, 0)])
    m = trimesh.creation.extrude_polygon(polygon, 30)
      
def draw():
    background(0)
    translate(width /2, height / 2)
    rotate_x(QUARTER_PI)
    rotate_y(radians(mouse_x))
    for i, face in enumerate(m.faces):
        fill(palette[i % 12])
        with begin_closed_shape():
            vertices([m.vertices[v] for v in face])
  ```

@hx2A, would it be too hard to make `vertices()` accept a generator so we could avoid building that list?

**Comment by hx2A** [comment]
> @hx2A, would it be too hard to make vertices() accept a generator so we could avoid building that list?

No, this is possible! I can do something like this:

```python
isinstance(foo, types.GeneratorType)
```

to test if something is a generator. Using that code, I can convert a generator to a list, and then pass that list to Java.

**Comment by villares** [comment]
![image](https://user-images.githubusercontent.com/3694604/216726738-03cfe40d-7a8a-4f09-8265-0b4acf43137d.png)

More experiments... https://github.com/villares/sketch-a-day/blob/main/2023/sketch_2023_02_03/sketch_2023_02_03.py

**Reply by hx2A** [reply]
Nice work! You are getting really good at working with fonts!

**Reply by villares** [reply]
I'm preparing for a fun activity at Sesc making festive "Carnival Masks with digital fabrication" and we plan to use some decorative dingbat fonts as source of elements, so, this was a good pretext to explore something I always wanted to get working on py5 (things that I used to know how to do in old Processing).

**Reply by hx2A** [reply]
Neat. Fonts can go beyond mere text, and be useful sources for unique shapes. I've gotten a lot of use out of snowflake fonts, for example.
